code/Items/ItemFilter.cs

```diff
@@ -11,36 +11,36 @@ public class ItemFilter
         /// <summary>
         /// Not authorized items
         /// </summary>
-        public HashSet<Item> Blacklist = new();
+        public HashSet<string> Blacklist = new();
 
         /// <summary>
         /// Authorized items
         /// </summary>
-        public HashSet<Item> Whitelist = new();
+        public HashSet<string> Whitelist = new();
 
-        public bool IsAuthorized(Item item)
+        public bool IsAuthorized(string itemId)
         {
-            if (Whitelist.Contains(item)) return true;
-            if (Blacklist.Contains(item)) return false;
+            if (Whitelist.Contains(itemId)) return true;
+            if (Blacklist.Contains(itemId)) return false;
             return !(Whitelist.Count > 0);
         }
 
-        public void AddToWhitelist(Item item)
+        public void AddToWhitelist(string item)
         {
             Whitelist.Add(item);
         }
```

Comment:
When you upgrade a class like that , you can let the old way.
You can add by id or by item. The item use the AddToWhitelist(string item)